### PR TITLE
feat: disable shorts player at app startup

### PR DESCRIPTION
**Pull request by OxrxL**
RE: https://github.com/revanced/revanced-integrations/pull/144

## TODO:

- [x] Implement the patch in a better way

### CODE REVIEW

File: src/main/kotlin/app/revanced/patches/youtube/layout/startupshortsreset/patch/StartupShortsResetPatch.kt [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import org.jf.dexlib2.iface.instruction.ReferenceInstruction
+import org.jf.dexlib2.iface.instruction.TwoRegisterInstruction
+import org.jf.dexlib2.iface.instruction.formats.Instruction21c
+import org.jf.dexlib2.iface.reference.MethodReference
+import org.jf.dexlib2.iface.reference.StringReference
+
+@Patch
+@DependsOn([IntegrationsPatch::class, SettingsPatch::class])
+@Name("disable-startup-shorts-player")
+@Description("Disable shorts player invoke after app execution.")
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/startupshortsreset/patch/StartupShortsResetPatch.kt:32]:
```suggestion
@Description("Disable playing YouTube Shorts when launching YouTube.")
```

File: src/main/kotlin/app/revanced/patches/youtube/layout/startupshortsreset/patch/StartupShortsResetPatch.kt [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+@StartupShortsResetCompatibility
+@Version("0.0.1")
+class StartupShortsResetPatch : BytecodePatch(
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/startupshortsreset/patch/StartupShortsResetPatch.kt:35]: Please name this class `DisableShortsOnStartupPatch` to match with the patch name.

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]
This patch would need a bit of cleaning on the implementation side, after that it can be merged.

**Comment by OxrxL** [comment]
Refactored as well :)

File: src/main/kotlin/app/revanced/patches/youtube/layout/startupshortsreset/patch/DisableShortsOnStartupPatch.kt [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        val iPutBooleanRegister = (actionOpenShortsIstructions[iPutBooleanIndex] as TwoRegisterInstruction).registerA
+
+        actionOpenShortsMethod.addInstructions(
+            moveResultIndex + 1, """
+            invoke-static { }, Lapp/revanced/integrations/patches/DisableStartupShortsPlayerPatch;->disableStartupShortsPlayer()Z
+            move-result v1
+            if-eqz v1, :cond_startup_shorts_reset
+            if-nez v$moveResultRegister, :cond_startup_shorts_reset
+            const/4 v$iPutBooleanRegister, 0x0
+            const/4 v$moveResultRegister, 0x1
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/startupshortsreset/patch/DisableShortsOnStartupPatch.kt:71]: It shouldn't be necessary for all this code for 3 registers. Instead use an opcode pattern as an anchor in the instructions to find the registers in a better way.

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

File: src/main/kotlin/app/revanced/patches/youtube/layout/startupshortsreset/patch/DisableShortsOnStartupPatch.kt [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                false,
+                StringResource("revanced_startup_shorts_player_summary_on", "Shorts player is disabled at app startup"),
+                StringResource("revanced_startup_shorts_player_summary_off", "Shorts player is enabled at app startup")
+            )
+        )
+
+        val actionOpenShortsMethod = ActionOpenShortsFingerprint.result!!.mutableMethod
+        val actionOpenShortsIstructions = actionOpenShortsMethod.implementation!!.instructions
+
+        val parentResult = ActionOpenShortsFingerprint.result!!
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/startupshortsreset/patch/DisableShortsOnStartupPatch.kt:50]: You should resolve the fingerprint on another fingerprints result, not itself.

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

**Pull request closed without merging**